exp/controllers/azuremangedcontrolplane_reconciler.go

```diff
@@ -228,15 +233,38 @@ func (r *azureManagedControlPlaneReconciler) reconcileEndpoint(ctx context.Conte
 		return fmt.Errorf("expected containerservice ManagedCluster object")
 	}
 
+	if managedCluster.AgentPoolProfiles == nil || len(*managedCluster.AgentPoolProfiles) < 1 {
+		return errors.New("expected at least one agent pool on cluster")
+	}
+
+	vnetSubnetID := (*managedCluster.AgentPoolProfiles)[0].VnetSubnetID
+
+	var vnetName string
+	if vnetSubnetID != nil && *vnetSubnetID != "" {
+		vnetName, err = extractVnetFromSubnetID(*vnetSubnetID)
+		if err != nil {
+			return err
+		}
+	} else {
+		var clusterID string
+		clusterID, err = getClusterID(*managedCluster.DNSPrefix)
```

Comment:
@CecileRobertMichon I know this is a hack, but it's actually just the same as clusterID from aks-engine (I looked at the code)

Happy to remove this and rely on the vnet being populated, though (I can't imagine why it wouldn't be).

Reply:
If we don't expect vnetSubnetID to be empty, I would prefer we return an error if it is. Relying on the format of the clusterID which is defined in another repo and not officially documented by AKS is dangerous IMO.

Reply:
removed

Reply:
So...turns out this field won't be populated unless it's a user-provisioned vnet. Maybe we should forget about this dance and just pre-provision the vnet? There's no user-facing reliable way to fetch it I think (which honestly seems very odd).

Reply:
apparently Paul Miller is looking at fixing this up on the RP side, I'll probably re-work this PR and pre-provision the network instead 🙁

Reply:
Already mentioned to Cecile offline, but AKS will expose this property publicly for all node pools. We're waiting on that functionality to move forward here (~1 or 2 weeks to full release)